test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/TagBuilderTest.cs

```diff
@@ -169,5 +169,74 @@ public void ReadingInnerHtml_LeavesHasInnerHtmlFalse()
             Assert.False(tagBuilder.HasInnerHtml);
             Assert.NotNull(innerHtml);
         }
+
+        [Fact]
+        public void RenderStartTag_RendersExpectedStartTag()
```

Comment:
Add a test or two which mucks with the `TagBuilder`'s `TagRenderMode` and confirms the `RenderX()` methods (a) still work and (b) don't change how the original tag is rendered.